### PR TITLE
deal with malformed HTTP response headers

### PR DESCRIPTION
**Pull request by mmacia**
Hi, I'm using crystal to implement a client API and I face an error when server responds with a malformed  HTTP message.

Servers should respond something like:
```
HTTP/1.1 200 OK\r\n
```
but in my case, sometimes responds:
```
HTTP/1.1 200\r\n
```
This leads to an error in `HTTP::Response#from_io` multiple assigment. This MR makes the method more resilient.

### CODE REVIEW

**Comment by ysbaddaden** [comment]
What about manually scanning each component, and be resilient to a missing message, instead of adding a specific condition?

**Comment by mmacia** [comment]
I made the scanning using a RE. Better?

**Comment by mmacia** [comment]
I made scanning with a RE. Better?

**Comment by asterite** [comment]
I think it's better to split in 3 and then just use an empty string if the third component is missing. For this you won't be able to use `http_version, status_code, status_message = ...` and just do `pieces = line.split(3)`, etc.

**Comment by mmacia** [comment]
I changed the code to use split

**Comment by asterite** [comment]
Thanks! Now [add a spec](https://github.com/manastech/crystal/blob/master/spec/std/http/response_spec.cr) too, please. Instead of `pieces[2]` you should do `pieces[2]?`.

**Comment by jhass** [comment]
we already have a list of well known status messages somewhere, shouldn't we try to fix the missing message if we allow that case at all? I don't feel good with having all user code using the message to effectively reimplement handling this case.

**Comment by mmacia** [comment]
I add the spec.
@asterite I don't get the point to use `pieces[2]?`. If `pieces[2]` is no `nil` I have to apply `chomp` function to rid of special characters.

I'am agree with @jhass, in that line I can substitute the empty string with `self.default_status_message_for(status_code)` as I did in previous commits to provide at least a valid status message.

What do you think?

**Comment by ysbaddaden** [comment]
`pieces[2]` will raise if there is no space after the 200, which we want to avoid (the original `a,b,c = line.split` raised for this very reason).

File: src/http/response.cr [hunk 1]
```diff
@@ -91,9 +91,11 @@ class HTTP::Response
   def self.from_io(io, ignore_body = false, &block)
     line = io.gets
     if line
-      http_version, status_code, status_message = line.split(3)
-      status_code = status_code.to_i
-      status_message = status_message.chomp
+      pieces = line.split(3)
+      http_version = pieces[0]
+      status_code = pieces[1].to_i
+      status_message = pieces[2]? ? pieces[2].chop : ""
```
Comment (asterite) [inline src/http/response.cr]: It should be `chomp` instead of `chop`

**Comment by mmacia** [comment]
Let me know if its OK now.

I think that would be fine if we try to provide a valid status message based on given status code:

```ruby
status_message = pieces[2]? ? pieces[2].chomp : self.default_status_message_for(status_code)
```

What do you think?

**Comment by asterite** [comment]
Empty string is good for now, I don't think the status message is something that you use in code, it's just informative.

**Comment by asterite** [comment]
Oops, should have asked for a squash...

**Pull request merged**